src/networking/RoleService.ts

```diff
@@ -65,14 +65,27 @@ export default class RoleService extends ModelService {
     ) as Promise<Role[]>;
   }
 
-  async getCurrentRoles(): Promise<Role[]> {
+  async getMostRecentCouncilMemberRoleBySeat(seatId: string): Promise<Role> {
     const populatePersonRef = new Populate(COLLECTION_NAME.Person, REF_PROPERTY_NAME.RolePersonRef);
 
     const networkQueryResponse = this.networkService.getDocuments(
       COLLECTION_NAME.Role,
-      [where("end_datetime", WHERE_OPERATOR.eq, null)],
+      [
+        where(
+          REF_PROPERTY_NAME.RoleSeatRef,
+          WHERE_OPERATOR.eq,
+          doc(NetworkService.getDb(), COLLECTION_NAME.Seat, seatId)
+        ),
+        limit(1),
+        orderBy("end_datetime", ORDER_DIRECTION.desc),
```

Comment:
I don't think `role.end_datetime` is ever null. If you sort by that property in seattle-staging, there are no documents with a null end_datetime. The current roles now have real dates. 

Sorting by a property should return documents that have that property as null. I assume that they are at the end of a list sorted in descending order, though. So, if there ARE null `end_datetime` roles, they would be at the end of this list which may be a problem.  @JacksonMaxfield care to weigh in?

Reply:
Yeah, I haven't see a null end_datetime yet, but it's a possibility according to https://github.com/CouncilDataProject/cdp-backend/blob/main/cdp_backend/database/models.py#L199 (not a required field)

Reply:
I think this is currently the most accurate query we have to get the full council. Maybe we correct if its starts returning accurate results? When did the `end_datetime` get populated, if it used to be `null`?

Reply:
>When did the end_datetime get populated, if it used to be null?


I think for all the instances we have, end_datetime always had a date.

I think Jackson allowed for "null" end_datetime because there is a possibility of an ongoing role with no end_datetime?

Reply:
> I think for all the instances we have, end_datetime always had a date.
> 
> I think Jackson allowed for "null" end_datetime because there is a possibility of an ongoing role with no end_datetime?

Yep. This is true. It may suck but is it possible to do two queries and set-addition the returns?

One query for most recent role by end datetime (like you currently have) and one query for roles with no end_datetime and then set addition them?

Reply:
My only concern is that the queries for roles with null `end_datetime` does leave us in a slightly uncertain state: we are inferring that those null roles found are the most recent, but we are not certain.

I do like the set-addition query better than our currently-deployed solution though, as it is based off of seats (so we won't NO SPONSOR entities). I will implement it.